tests/devicetree_test.py

```diff
@@ -476,3 +431,123 @@ def test_get_related_disks(self):
+        self.assertTrue(self.sda in self.tree.devices)
+        self.assertTrue(self.sdb in self.tree.devices)
+        self.assertTrue(self.sdc in self.tree.devices)
+        self.assertTrue(self.mpatha in self.tree.devices)
+
+    def test_exclusive_disks_multipath_3(self):
+        # some disks exclusive -> mpath should be hidden
+        self.tree.exclusive_disks = ["sda", "sdb"]
+        with patch.object(self.tree, "hide") as hide:
+            self.tree._hide_ignored_disks()
+            hide.assert_any_call(self.mpatha)
+            hide.assert_any_call(self.sdc)
+
+        # verify that hide works as expected
+        self.tree._hide_ignored_disks()
+        self.assertTrue(self.sda in self.tree.devices)
+        self.assertTrue(self.sdb in self.tree.devices)
+        self.assertFalse(self.sdc in self.tree.devices)
+        self.assertFalse(self.mpatha in self.tree.devices)
+
+    def test_ignored_disks_multipath_1(self):
+        # mpatha ignored -> disks should be hidden
+        self.tree.ignored_disks = ["mpatha"]
+        self.tree.exclusive_disks = []
+
+        with patch.object(self.tree, "hide") as hide:
+            self.tree._hide_ignored_disks()
+            hide.assert_any_call(self.mpatha)
+            hide.assert_any_call(self.sda)
+            hide.assert_any_call(self.sdb)
+            hide.assert_any_call(self.sdc)
+
+        self.tree._hide_ignored_disks()
+        self.assertFalse(self.sda in self.tree.devices)
+        self.assertFalse(self.sdb in self.tree.devices)
+        self.assertFalse(self.sdc in self.tree.devices)
+        self.assertFalse(self.mpatha in self.tree.devices)
+
+    def test_ignored_disks_multipath_2(self):
+        # all disks ignored -> mpath should be hidden
+        self.tree.ignored_disks = ["sda", "sdb", "sdc"]
+        self.tree.exclusive_disks = []
+
+        with patch.object(self.tree, "hide") as hide:
+            self.tree._hide_ignored_disks()
+            hide.assert_any_call(self.mpatha)
+            hide.assert_any_call(self.sda)
+            hide.assert_any_call(self.sdb)
+            hide.assert_any_call(self.sdc)
+
+        self.tree._hide_ignored_disks()
+        self.assertFalse(self.sda in self.tree.devices)
+        self.assertFalse(self.sdb in self.tree.devices)
+        self.assertFalse(self.sdc in self.tree.devices)
+        self.assertFalse(self.mpatha in self.tree.devices)
+
+    def test_ignored_disks_multipath_3(self):
+        # some disks ignored -> mpath should be hidden
+        self.tree.ignored_disks = ["sda", "sdb"]
+        self.tree.exclusive_disks = []
+
+        with patch.object(self.tree, "hide") as hide:
+            self.tree._hide_ignored_disks()
+            hide.assert_any_call(self.sda)
+            hide.assert_any_call(self.sdb)
+
+        self.tree._hide_ignored_disks()
+        self.assertFalse(self.sda in self.tree.devices)
+        self.assertFalse(self.sdb in self.tree.devices)
+        self.assertTrue(self.sdc in self.tree.devices)
```

Comment:
I agree, but on the other hand this is exactly what user wants in this case. And exclusive disks also behave like this.

Reply:
I've change the behaviour to ignore requests to hide only some of the multipath disks -- ignoring only "sda" will result in both sda and sdb to be visible (and of course the multipath device will be visible too) and a warning will be logged.

Reply:
Stepping back a little bit to think about what is the best behavior for blivet in this situation...

If anything I think we should err on the side of protecting the user's data. I'm not comfortable opening users to accidentally overwriting their shared/SAN storage because they only hid a subset of the paths (although I think most people would be doing this by wwid/wwn/serial rather than by name, vastly reducing the odds of this situation occurring at all). If possible, we could automatically hide the mpath along with all of the other members when someone hides one of them. A slightly less aggressive approach would be to allow only hiding some of the paths/members, but then raise an exception if they try to allocate partitions, schedule actions, or execute actions while the tree is in an inconsistent state. What do you think? I can't really think of any downside to being a bit more careful/considerate in this case. Can you?

Reply:
Ok, I've changed the warning to an exception. I agree it's safer and it really doesn't make sense to ignore just part of the multipath or dmraid device so raising an exception shouldn't be a problem.

I', not sure if the exception message makes sense, feel free to suggest a better one :-)